### PR TITLE
Choose double tap

### PR DESCRIPTION
**Pull request by Huddie**
Allowing users to choose the reaction that occurs when double tapping on a comment

### CODE REVIEW

File: Classes/Settings/SettingsViewController.swift [hunk 1]
```diff
@@ -259,3 +268,69 @@ NewIssueTableViewControllerDelegate {
...
+  @objc private func onHeart() {
+   updateDefaultReaction(.heart)
+  }
+  
+  func updateDefaultReaction(_ reaction: ReactionContent)
+  {
+    UserDefaults.setDefault(reaction: reaction)
+    defaultReactionLabel.text = reaction.emoji
+  }
+}
```
Comment (rnystrom) [inline Classes/Settings/SettingsViewController.swift]: Fine merging this as-is, but if we're going to put stuff in an `extension`, we might as well put it in another file. We don't really follow the trend of making extensions for everything in this project.

File: Classes/Settings/SettingsViewController.swift [hunk 2]
```diff
@@ -259,3 +268,69 @@ NewIssueTableViewControllerDelegate {
...
+      (ReactionContent.laugh.emoji, #selector(SettingsViewController.onLaugh)),
+      (ReactionContent.hooray.emoji, #selector(SettingsViewController.onHooray)),
+      (ReactionContent.confused.emoji, #selector(SettingsViewController.onConfused)),
+      (ReactionContent.heart.emoji, #selector(SettingsViewController.onHeart))
+    ]
+    
+    let menu = UIMenuController.shared
+    menu.menuItems = actions.map { UIMenuItem(title: $0.0, action: $0.1) }
+    menu.setTargetRect(defaultReactionLabel.frame, in: setDefaultReaction)
+    menu.setMenuVisible(true, animated: trueUnlessReduceMotionEnabled)
```
Comment (rnystrom) [inline Classes/Settings/SettingsViewController.swift]: Using `UIMenuController` to pick a selection in Settings feels really weird. Could we instead create a new `UITableViewController` that has all of the emoji options, and you select a cell?

Comment (Huddie) [reply Classes/Settings/SettingsViewController.swift]: @rnystrom Context Menu type thing or you want a whole new view controller?

File: Classes/Settings/SettingsViewController.swift (hunk 1 above)
Comment (Huddie) [reply Classes/Settings/SettingsViewController.swift]: I'll just remove the extension no biggy.

File: Classes/Settings/SettingsViewController.swift (hunk 2 above)
Comment (rnystrom) [reply Classes/Settings/SettingsViewController.swift]: While new view controller. Like how on settings sub menus are just a simple list:

GitHawk Upload by rnystrom

**Comment by Huddie** [comment]
This has been updated to the following! Minor wording changes were made after these screenshots (Leave this by double -> Leave this when double) either way could work, let me know quick fix. And Smiley was changed to Laugh for the emoji name.
 
img_1730

img_1731

**Comment by Huddie** [comment]
Cells have switched to styledTextCells to match the rest of settings page

**Comment by rnystrom** [comment]
Couple more UX things:

- Let's change the title of the cell to "Double Tap Reaction"
- Can we just make it part of the GitHawk section? I don't think we'll ever have more options under "Reactions"
- Let's remove the "Leave this by double tapping..." part. If we title the option "Double Tap Reaction" that should be pretty explanatory
- How about instead of "None" we say "Disabled" or "Off"?

**Comment by Huddie** [comment]
@rnystrom Ya I don't mind having it under GitHawk I think that the issues I ref above just spoke about the fact that settings arent really broken up that much and it could use a few more sections to make it easier to navigate. The GitHawk section seems to contain settings that pertain to GitHawk as an app (Report a bug, view source code, write a review) As well as settings that pertain to the specfic users app (Badge Unread, Read on Open, and if reactions, reactions) So half are about customizing your version of GitHawk half are about GitHawk itself. Maybe we break these up into 2 sections. GitHawk and Customize?

**Comment by Huddie** [comment]
Ill shift it into GitHawk section and I guess breaking the section up can become an issue or really it should pick up from the already created issue that discusses it

**Comment by Huddie** [comment]
@rnystrom Okay those UX fixes + I changed how none/disabled enabled worked.

**Comment by Huddie** [comment]
New enable/disable UI When enabled by HuddieWhen disabled by Huddie

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Huddie** [comment]
Also addresses #1970

**Comment by rnystrom** [comment]
Looks awesome! I'm about to cut + ship the latest version. Going to hold on landing this until we ship so we have enough time to dogfood this.

**Comment by Huddie** [comment]
Awwww man! Haha sounds good. Looking forward to the next release!

**Comment by BasThomas** [comment]
Looks amazing @Huddie! 😍

**Comment by Huddie** [comment]
Thanks @BasThomas!

**Comment by rnystrom** [comment]
Gonna take it and give it a whirl 😄

**Pull request merged**